lib/rubocop/cop/style/redundant_string_escape.rb

```diff
@@ -42,10 +42,9 @@ class RedundantStringEscape < Base
         MSG = 'Redundant escape of %<char>s inside string literal.'
 
         def on_str(node)
-          return if node.parent&.regexp_type? || node.parent&.xstr_type?
+          return if node.parent&.regexp_type? || node.parent&.xstr_type? || character_literal?(node)
```

Comment:
If https://github.com/rubocop/rubocop-ast/pull/242 becomes available, it can be rewritten as:

```suggestion
          return if node.parent&.regexp_type? || node.parent&.xstr_type? || node.character_literal?
```

Reply:
That'd be nice - we have several checks for char literals in the codebase that can be simplified. And one cop that discourages them. :-)